Extension/src/LanguageServer/lmTool.ts

```diff
@@ -44,6 +47,84 @@ const knownValues: { [Property in keyof ChatContextResult]?: { [id: string]: str
     }
 };
 
+function formatChatContext(chatContext: ChatContextResult): void {
+    type KnownKeys = 'language' | 'standardVersion' | 'compiler' | 'targetPlatform' | 'targetArchitecture';
+    for (const key in knownValues) {
+        const knownKey = key as KnownKeys;
+        if (knownKey && knownValues[knownKey] && chatContext[knownKey]) {
+            chatContext[knownKey] = knownValues[knownKey][chatContext[knownKey] as string] || chatContext[knownKey];
+        }
+    }
+}
+
+export interface CppContext {
+    language: string;
+    standardVersion: string;
+    compiler: string;
+    targetPlatform: string;
+    targetArchitecture: string;
+    compilerArguments: string;
+}
+
+const matchNothingRegex = /(?!.*)/;
+
+// To be updated after A/B experiments, math nothing for now.
+const defaultCompilerArgumentFilters: { [id: string]: RegExp } = {
+    MSVC: matchNothingRegex,
+    Clang: matchNothingRegex,

```

Comment:
One thing to consider is that on macOS the default compiler is clang, and `/usr/bin/gcc` / `/usr/bin/g++` are actually symbolic links to clang, not gcc.  Performing a regex match on the compiler path may not be a reliable way to determine what compiler is being referred to. On the native side, we 'query' non-CL compilers (gcc or clang), and will use system defines to determine whether the compiler is clang or gcc.  (There could also be other details that could be determined based on detected system defines, that may potentially be worth surfacing to AI in the future).  I'd suggest adding plumbing on the native side to pass back the 'compiler type' (with the rest of the TU context) instead of relying on pattern matching against compiler paths.

Reply:
These regex are for matching the compiler arguments to be sent to the Copilot, these are not for detecting the compiler types.

Reply:
I see.  FYI, the source of "compiler" here appears to be coming from the `IntelliSenseMode` on the native side, only.  It's not quite correct to directly associate that with compiler arguments.  For instance, clang has a compiler, `clang-cl.exe`, which is clang, but accepts MSVC command line arguments (as well as some clang arguments not accepted by MSVC).  The user will get a (i.e.) `windows-clang-x64` intelliSense-mode, despite all of their arguments being CL arguments.

Reply:
Thanks, Colen, I didn't know that, I think we can configure the regex to match the arguments from both scenarios for Clang so that we can cover `windows-clang-x64` as well. So, I think we are good.